lib/timeline.js

```diff
@@ -0,0 +1,107 @@
+class Timeline {
+  constructor(client) {
+    this.client = client
+  }
+
+  createEventType(applicationId, userId, data, cb) {
+    return this.client._request(
+      {
+        method: 'POST',
+        path: `/integrations/v1/${applicationId}/timeline/event-types?userId=${userId}`,
+        body: data,
+      },
+      cb,
+    )
+  }
+
+  updateEventType(eventId, applicationId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createEventTypeProperty(eventId, applicationId, userId, data, cb) {
+    const parameters = {
+      method: 'POST',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties?userId=${userId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  updateEventTypeProperty(eventId, applicationId, propertyId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties/${propertyId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createTimelineEvent(applicationId, eventTypeId, data, cb) {
+    if (
+      !this.client.clientId ||
```

Comment:
I'm not sure this is true. I've been able to use the Timeline API with only an oauth token. So if we wanted to raise an error we could do:

```js
if (!this.client.auth) {
  return Promise.reject(new Error(...))
}
```
But I think we should be ok not making this check here.